File: HackPDM_CSharp/OdooClient/OdooModels/Models/HpCategory.cs
//using static System.Net.Mime.MediaTypeNames;

namespace HackPDM
{
    public class HpCategory : HpBaseModel<HpCategory>
    {
        internal readonly string[] usualExcludedFields = [];
        public string name;
        public string cat_description;
        public bool track_version;
        public bool track_depends;

        public HpCategory() { }
        public HpCategory(
            string name,
            string cat_description = "CAD files are versioned and have dependencies",
            bool track_version = true,
            bool track_depends = true)
        {
            this.name = name;
            this.cat_description = cat_description;
            this.track_version = track_version;
            this.track_depends = track_depends;
        }
		public override string ToString()
		{
			return name;
		}
	}
}

File: Odoo/addons/hackpdm/models/hp_category.py
from odoo import models, fields, api

class hp_category(models.Model):
    _name = 'hp.category'
    _description = 'category'
    _inherit = 'hp.common.model'

    name = fields.Char(
        string='category name',
        index='trigram',
        required=True,
    )
    cat_description = fields.Char(
        default='CAD files are versioned and have dependencies.',
        string='category description',
    )
    #parent_path = fields.Char(index=True, unaccent=False)

    track_version = fields.Boolean(
        default=True,
        string='track version',
    )
    track_depends = fields.Boolean(
        default=True,
        string='track depends',
    )


class hp_category_property(models.Model):
    _name = 'hp.category.property'
    _description = 'hp category property'
    _inherit = 'hp.common.model'

    cat_id = fields.Many2one(
        comodel_name='hp.category',
        string='category id',
    )
    prop_id = fields.Many2one(
        comodel_name='hp.property',
        string='property',
    )
